corehq/apps/reports/standard/deployments.py

```diff
@@ -234,6 +247,38 @@ def process_rows(self, users, fmt_for_export=False):
             ])
         return rows
 
+    def process_users(self, users, fmt_for_export=False):
+        rows = []
+        for user in users[self.pagination.start:self.pagination.end]:
+            rows.append([
+                user_display_string(user.user_dim.username,
+                                    user.user_dim.first_name,
+                                    user.user_dim.last_name),
+                _fmt_date(user.last_form_submission_date, fmt_for_export),
+                _fmt_date(user.last_sync_log_date, fmt_for_export),
+                self.get_app_name(user.app_dim.app_id) or '---',
+                user.last_form_app_build_version,
+                user.last_form_app_commcare_version
+            ])
+
+    def get_sql_sort(self):
+        res = []
+        #the NUMBER of cols sorting
+        sort_cols = int(self.request.GET.get('iSortingCols', 0))
+        if sort_cols > 0:
+            for x in range(sort_cols):
+                col_key = 'iSortCol_%d' % x
+                sort_dir = self.request.GET['sSortDir_%d' % x]
+                col_id = int(self.request.GET[col_key])
+                col = self.headers.header[col_id]
+                if col.sql_col is not None:
+                    res = col.sql_col
+                    if sort_dir == 'desc':
```

Comment:
Would it make sense to raise an error if `sort_dir` is not one of the expected values (`asc` or `desc`)? What if someone makes a request with `sSortDir_0=DESC` or `sSortDir_0=dsc`? Currently those would have the same outcome as `sSortDir_0=asc`.